src/broker/cas_network.c

```diff
@@ -581,13 +605,13 @@ read_buffer (SOCKET sock_fd, char *buf, int size)
 	{
 #endif /* ASYNC_MODE */
 	  /* RECEIVE NEW REQUEST */
-	  read_len = READ_FROM_SOCKET (sock_fd, buf, size);
+	  read_len = READ_FROM_NET (sock_fd, buf, size);
 #if defined(ASYNC_MODE)
 	}
     }
 #endif /* ASYNC_MODE */
 
-  if (read_len <= 0)
+  if ((!ssl_client && read_len <= 0) || (ssl_client && read_len < 0))
```

Comment:
When ssl_client is true, Is that read_len is 0 no problem ?
SSL_read() success if return value great than 0 (> 0).

Reply:
I misunderstood the SSL_read manual. 
You are right. SSL_read returns 0 means the connection was closed, and error.
I will fix it. Thank you for your comment.

Reply:
It's my pleasure.